File: ruby/robot-name/robot_name.rb
class Robot
 attr_reader :name

  def initialize
    generate
  end

  def generate
    pre = [*'A'..'Z'].sample(2).join
    suff = rand(100..999).to_s
    @name = pre + suff
  end

  def reset
    generate
  end

end

# class Robot
#   attr_reader :name
#
#   def initialize
#     set_name
#   end
#
#   def set_name
#     @name = (0...2).map { (65 + rand(26)).chr }.join + (rand 100..999).to_s
#   end
#
#   def reset
#     set_name
#   end
#
# end


also, what is ||=
example def name
          @name ||= generate_name
        end

File: javascript/robot-name/robot-name.js
var Robot = function(){
  this.name = this.generateName();
};

Robot.prototype.name = function(){
  return this.name;
};

Robot.prototype.generateName = function(){
  var name = "";

  for (var i = 0; i < 2;i++){
    name += String.fromCharCode(97 + Math.floor(Math.random() * 26)).toUpperCase();
  }
  for (var y = 0; y < 3;y++){
    name += Math.floor(Math.random() * 9) + 1;
  }

  if (names.indexOf(name) === -1) {names.push(name); return name; }
  return this.generateName();
};

Robot.prototype.reset = function(){
  this.name = this.generateName();
};

var names = [];
module.exports = Robot;
